File: Java/1. Two Sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> lookup = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            lookup.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (lookup.containsKey(complement) && lookup.get(complement) != i) {
                return new int[] { i, lookup.get(complement) };
            }
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: LeetCode/1. Two Sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	pair := make(map[int]int)

	l := len(nums)
	for i := 0; i < l; i++ {
		if j, ok := pair[nums[i]]; ok {
			return []int{j, i}
		}
		pair[target-nums[i]] = i
	}
	return []int{}
}

func main() {
	fmt.Println(twoSum([]int{3, 2, 4}, 6))
}
